static/app/views/settings/account/notifications/notificationSettingsByType.tsx

```diff
@@ -320,173 +324,152 @@ class NotificationSettingsByTypeV2 extends DeprecatedAsyncComponent<Props, State
     });
     const fields: Field[] = [defaultField];
     return fields;
-  }
-
-  getLinkedOrgs = (provider: SupportedProviders): OrganizationSummary[] => {
-    const {organizations} = this.props;
-    const {identities, organizationIntegrations} = this.state;
-    const integrationExternalIDsByOrganizationID = Object.fromEntries(
-      organizationIntegrations
-        .filter(
-          organizationIntegration => organizationIntegration.provider.key === provider
-        )
-        .map(organizationIntegration => [
-          organizationIntegration.organizationId,
-          organizationIntegration.externalId,
-        ])
-    );
-
-    const identitiesByExternalId = Object.fromEntries(
-      identities.map(identity => [identity?.identityProvider?.externalId, identity])
-    );
-
-    return organizations.filter(organization => {
-      const externalID = integrationExternalIDsByOrganizationID[organization.id]!;
-      const identity = identitiesByExternalId[externalID];
-      return !!identity;
-    });
   };
 
-  getUnlinkedOrgs = (provider: SupportedProviders): OrganizationSummary[] => {
-    const linkedOrgs = this.getLinkedOrgs(provider);
-    const {organizations} = this.props;
-    return organizations.filter(organization => !linkedOrgs.includes(organization));
-  };
-
-  handleRemoveNotificationOption = async (id: string) => {
-    await this.api.requestPromise(`/users/me/notification-options/${id}/`, {
-      method: 'DELETE',
-    });
-    this.setState(state => {
-      const newNotificationOptions = state.notificationOptions.filter(
-        option => !(option.id.toString() === id.toString())
+  const removeNotificationMutation = useMutation({
+    mutationFn: (id: string) =>
+      fetchMutation(['DELETE', `/users/me/notification-options/${id}/`]),
+    onSuccess: (_, id) => {
+      setApiQueryData<NotificationOptionsObject[]>(
+        queryClient,
+        notificationOptionsQueryKey(notificationType),
+        currentNotificationOptions => {
+          return currentNotificationOptions?.filter(
+            option => !(option.id.toString() === id.toString())
```

Comment:
```suggestion
            option => option.id.toString() !== id.toString()
```

Seems like a very complex way to say not equal lol

Reply:
indeed, I didn’t even see that 👍